content/articles/hashing-in-data-structure/index.md

```diff
@@ -0,0 +1,299 @@
+
+        //insert in new table
+        for(int i=0;i<oldts;i++){
+            node* head = oldarr[i];
+            while(head){
+                insert(head->name,head->value);
+                head=head->next;
+            }
+        }
+        delete []oldarr;
+
+
+    }
+
+    public:
+    hashmap(int s=7){
+        arr = new node*[s];
+        ts=s;
+        cs=0;
+        for(int i=0;i<s;i++){
+            arr[i]=NULL;
+        }
+
+    }
+
+    void insert(string key, int data){
+        int i=hashfn(key);
+        node* n=new node(key,data);
+        n->next=arr[i];
+        arr[i]=n;
+        cs++;
+
+        if(cs/(1.0*ts)  > 0.6){
+            reHash();
+        }
+
+    }
+
+    node* search(string key){
+        int i=hashfn(key);
+        node*head= arr[i];
+        while(head){
+            if(head->name==key){
+               return head;
+                break;
+            }
+            head=head->next;
+        }
+        if(head==NULL){
+            cout<<"not exist";
+        }
+        return NULL;
+    }
+
+    void print(){
+        for(int i=0;i<ts;i++){
+            node* head= arr[i];
+            while(head){
+                cout<<head->name<<"-->"<<head->value;
+                head=head->next;
+                 cout<<endl;
+            }
+
+        }
+    }
+
+    int& operator[](string key){
+        node* ans=search(key);
+        if(ans){
+            return ans->value;
+        }
+        else{
+            int agrbagevalue;
+            insert(key,agrbagevalue);
+            ans= search(key);
+            return ans->value;
+        }
+
+    }
+
+    void Delete(string key){
+        int i=hashfn(key);
+        node* head= arr[i];
+        node* trail=NULL;
+
+            while(head){
+
+                    //first
+                    if(head->name==key && trail==NULL){
+                        arr[i]=NULL;
+                        delete head;
+                    }
+                    //in end
+                    if(head->name==key && trail!=NULL && head->next==NULL){
+                        trail->next=NULL;
+                        delete head;
+                    }
+                    //mid
+                    if(head->name==key && head->next!=NULL){
+                        node*ptr =head;
+                        head=head->next;
+                        delete ptr;
+                        arr[i]=head;
+
+                    }
+                    trail =head;
+                    head=head->next;
+            }
+        return;
+    }
+
+};
+
+int main(){
+    hashmap h;
+    h.insert("Alphonso_Mango",100);
+    h.insert("Kiwi",150);
+    h.insert("Banana",200);
+    h.insert("WaterMelon",180);
+    h.print();
+
+    node* x=  h.search("Kiwi");
+    cout<<"SEARCH RESULT"<<endl;
+    cout<<x->name<<" "<<x->value;
+
+}
+
+```
+
+The complete execution of the above code can be found [here](https://replit.com/@Anubhavb11/HashMap#main.cpp).
+=======
+
+#### Output
+
+![output](./Capture.PNG)
+
```

Comment:
Please explain in detail these output results. Also, it's good practice to end an article with a "Conclusion" section which wraps up what your article is about.